android/src/main/java/com/swmansion/reanimated/NodesManager.java

```diff
@@ -202,7 +205,7 @@ private void stopUpdatingOnAnimationFrame() {
   }
 
   public void performOperations() {
-    if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
+    if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED && mNativeProxy != null) {
       mNativeProxy.performOperations();
     } else if (!mOperationsInBatch.isEmpty()) {
```

Comment:
Is any benefit of this syntax? 🤔

Reply:
It's not only about syntax but also about the semantics. Assume we're on Fabric so `BuildConfig.IS_NEW_ARCHITECTURE_ENABLED` is true. If `mNativeProxy == null`, the condition fails and we will go to `else if` branch which is supposed to be Paper-only.
```java
    if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED && mNativeProxy != null) {
      mNativeProxy.performOperations();
    } else if (!mOperationsInBatch.isEmpty()) {
      //
    }
```
I think we don't populate `mOperationsInBatch` on Fabric so effectively this will do nothing but ideally we don't want to touch `mOperationsInBatch` at all on Fabric.
However, if we have:
```java
    if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
      if (mNativeProxy != null) {
        mNativeProxy.performOperations();
      }
    } else if (!mOperationsInBatch.isEmpty()) {
```
Then if `mNativeProxy == null` we simply do nothing and we don't go into `else if` branch.

Reply:
ok, make sense 🫡 fixed in https://github.com/software-mansion/react-native-reanimated/pull/5016/commits/2111debc0b6d38c6a4ab5e714f4c32c559a3d965